contracts/signature-bridge/src/contract.rs

```diff
@@ -81,37 +89,33 @@ fn admin_set_resource_with_signature(
     let mut data: Vec<u8> = Vec::new();
     data.extend_from_slice(&msg.resource_id);
     data.extend_from_slice(&msg.function_sig);
-    data.extend_from_slice(&element_encoder(&msg.nonce.to_le_bytes()));
+    data.extend_from_slice(&msg.nonce.to_be_bytes());
     data.extend_from_slice(&msg.new_resource_id);
     data.extend_from_slice(msg.handler_addr.as_bytes());
     data.extend_from_slice(msg.execution_context_addr.as_bytes());
 
-    if !signed_by_governor(deps.branch(), &data, &msg.sig, state.governor.as_str())? {
+    if !signed_by_governor(deps.branch(), &data, &msg.sig, &state.governor)? {
         return Err(ContractError::Std(StdError::GenericErr {
             msg: "Invalid sig from governor".to_string(),
         }));
     }
 
-    if msg.nonce != state.proposal_nonce + 1048 {
+    if msg.nonce <= state.proposal_nonce || state.proposal_nonce + 1048 < msg.nonce {
         return Err(ContractError::InvalidNonce);
     }
 
-    let func_sig = Keccak256::hash(
+    let func_sig = keccak_256(
         b"adminSetResourceWithSignature(bytes32,bytes4,uint32,bytes32,address,address,bytes)",
-    )
```

Comment:
Why you are using Ethereum function signature here? if there is no use of it you can do the same we do in Substrate, we just use "0x00000000" (4 bytes all zeros).

Reply:
Ok, will go for update as mentioned.